File: backend/entity/promotion.go
// entity/promotion.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Promotion struct {
	gorm.Model
	Title        string    `json:"title"`
	Description  string    `json:"description"`
	DiscountValue int      `json:"discount_value"` // คิดเป็นจำนวนเงินหรือเปอร์เซ็นต์แล้วแต่กติกา
	StartDate    time.Time `json:"start_date"`
	EndDate      time.Time `json:"end_date"`
	PromoName    string    `json:"promoname"`

	CreatedBy uint  `json:"createdby"`
	Creator   *User `gorm:"foreignKey:CreatedBy" json:"creator,omitempty"`

	OrderPromotions []OrderPromotion `gorm:"foreignKey:PromotionID" json:"order_promotions,omitempty"`
}

File: frontend/src/interfaces/Promotion.ts
import type { User } from "./User";
import type { OrderPromotion } from "./OrderPromotion";

export interface Promotion {
  ID: number;
  title: string;
  description: string;
  discount_value: number;
  start_date: string;   // ISO datetime
  end_date: string;     // ISO datetime
  promoname: string;

  createdby: number;
  creator?: User;

  order_promotions?: OrderPromotion[];
}

export interface CreatePromotionRequest {
  title: string;
  description: string;
  discount_value: number;
  start_date?: string;
  end_date?: string;
  promoname: string;
  createdby: number;
}

export interface UpdatePromotionRequest {
  ID: number;
  title?: string;
  description?: string;
  discount_value?: number;
  start_date?: string;
  end_date?: string;
  promoname?: string;
}
